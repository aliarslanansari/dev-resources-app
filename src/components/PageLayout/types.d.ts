declare interface TPageLayoutProps {
  children: NotNull<React.ReactNode>
}

export declare type TPageLayoutPropsType = TPageLayoutProps
