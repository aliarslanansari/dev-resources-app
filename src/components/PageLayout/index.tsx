import * as React from 'react'
import { TPageLayoutPropsType } from './types'
import { makeStyles, Theme } from '@material-ui/core/styles'

const usePageLayoutStyles = makeStyles((theme: Theme) => ({
  root: {
    height: '100vh',
    backgroundColor: theme.palette.primary.dark,
    backgroundSize: 'cover',
    display: 'flex',
    flex: 1,
    overflow: 'hidden',
    flexDirection: 'column'
  }
}))

export default function PageLayout(props: TPageLayoutPropsType) {
  const classes = usePageLayoutStyles()

  return (
    <div data-testid='pageLayout' className={classes.root}>
      {props.children}
    </div>
  )
}
