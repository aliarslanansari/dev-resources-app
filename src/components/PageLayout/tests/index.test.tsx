import { render } from '@testing-library/react'
import PageLayout from '../index'

describe('<PageLayout />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = render(
      <PageLayout>
        <h1>Hello</h1>
      </PageLayout>
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should contain 1 PageLayout component', () => {
    const { getAllByTestId } = render(
      <PageLayout>
        <h1>Hello</h1>
      </PageLayout>
    )
    expect(getAllByTestId('pageLayout').length).toBe(1)
  })
})
