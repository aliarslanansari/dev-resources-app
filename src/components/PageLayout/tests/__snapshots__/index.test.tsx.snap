// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLayout /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-root-1"
      data-testid="pageLayout"
    >
      <h1>
        Hello
      </h1>
    </div>
  </div>
</body>
`;
